config_service.proto

```diff
@@ -0,0 +1,64 @@
+syntax = "proto3";
+
+package org.hypertrace.agent.config;
+
+import "google/protobuf/wrappers.proto";
+
+option go_package = "github.com/hypertrace/goagent/config";
+option java_package = "org.hypertrace.agent.config";
+
+import "config.proto";
+
+service ConfigurationService {
+  // InitialConfiguration returns the initial configuration for an agent.
+  // It should be called by agents only if config file is not specified by user.
+  rpc InitialConfiguration(InitialConfigurationRequest) returns (InitialConfigurationResponse) {}
+
+  // UpdateConfiguration returns the configs that can be dynamically updated in agents
+  rpc UpdateConfiguration(UpdateConfigurationRequest) returns (UpdateConfigurationResponse) {}
+}
+
+// InitialConfigurationRequest has information about the agent
+message InitialConfigurationRequest {
```

Comment:
Not using `service_name` here as a user will anyways define it in env var or system property and they have higher preference so it will be updated by agents. Do we have any other use case for sending `service_name`?